### DISCUSSION TITLE
Installation fails on Docker in Synology NAS / Error when connecting to DB: SQLSTATE [HY000] [2002] Connection timed out

### DISCUSSION THREAD

**Discussion by Werfjes** [firefly-iii/firefly-iii#5137] [Firefly III]
I am running Firefly III latest docker version and db is running on the NAS (not on a Docker)

After running the image and accessing it via "port 3307" I get the following messages: 
**Error when connecting to DB: SQLSTATE [HY000] [2002] Connection timed out**
**Could not poll the database: SQLSTATE [HY000] [2002] Connection timed out**

Similar discussions / questions were made earlier (see links below), but it either no solution was provided, didn't work or I didn't understand it (probably the latter)  :
- https://github.com/firefly-iii/firefly-iii/discussions/4878
- https://github.com/firefly-iii/firefly-iii/discussions/5094
- https://github.com/firefly-iii/firefly-iii/discussions/5048

![FireFly](https://user-images.githubusercontent.com/91469589/135242269-cc9a5c92-f36f-4b08-bab4-c7b7462eba33.JPG)

Edit: added a log file
[fireflyiii-core1.csv](https://github.com/firefly-iii/firefly-iii/files/7251636/fireflyiii-core1.csv)

**Comment by JC5** [comment]
Is there a DB running on port 3307?

**Reply by Werfjes** [reply]
yes
![DB running](https://user-images.githubusercontent.com/91469589/135408843-2674fa93-bedd-4150-a0a2-be0f39d442d6.JPG)

**Reply by Werfjes** [reply]
Can somebody pls help me. Thx

**Reply by JC5** [reply]
Sorry I have no clue and know too little about Synology. I hope somebody else can help you debug!

**Reply by MacPaille** [reply]
I have also a Syno but running MariaDB in a container, it's working fine.
OK the DB is listening on port 3307 but is it allowed to get connections from external services (I mean not only localhost)? Also what is your IP configuration on the NAS and the container itself and what is your config on container level (the environment variables to set the DB information)

**Reply by Werfjes** [reply]
Hi MacPaille,

Thanks for you help.

Herewith the info

**Environment Variables** (Syno Docker)
DB_PORT: 3307
DB_HOST: 192.168.88.199

**IP address of Synology NAS**: 
192.168.88.199

**Maria DB10** (is not running in a container, but as an app in Synology)
Enable TCP/IP connection box is checked
Port 3307

**Export / Log file**
pls notice attachement

**Screenshot Terminal**
pls notice attachement

Please let me know if you need more info & and thanks in advance

[fireflyiii-core1.csv](https://github.com/firefly-iii/firefly-iii/files/7839714/fireflyiii-core1.csv)
![ScreenShot Terminal](https://user-images.githubusercontent.com/91469589/148788772-f4c4a7db-e10d-489d-8fc6-75bf81b30c50.JPG)

**Reply by MacPaille** [reply]
@Werfjes What is your container network config? Have you set it as a bridge?
It definitely gets a timeout because it can't reach the IP of your NAS on port 3307... now to make sure the connectivity works you should definitely go in SSH in the nas to play around with tcpdump command and validate there is something coming from the container...

I have no issue with my containers having the MariaDB in one and Firefly in another that has a link to the DB... Maybe you can try that setup too... Is it mandatory for you to have the firefly DB on your host and not in a container?

**Comment by LM3469** [comment]
Hey there, I have the exact same issue and did not find any way to resolve it yet.
It would be much appreciated if anyone could dig a little bit the issue we face to install firefly III on NAS Synology using Docker.
Thanks

**Comment by LM3469** [comment]
I have found a way to resolve my issue, it may work for you:

**Maria DB10 (is not running in a container, but as an app in Synology)**
Enable TCP/IP connection box is checked
Port = 3307
Make sure that MariaDB port remain close from incoming connexions except from your NAS

**Firefly DB and User created in MariaDB using Phpmyadmin**
user host = localhost

**Network of Firefly-iii container (Syno Docker)**
Use docker host

**Environment Variables of Firefly-iii container (Syno Docker)**
DB_PORT = 3307
DB_HOST = 127.0.0.1 (do not use the NAS IP or "localhost")

**Reply by Werfjes** [reply]
HI LM3469
Thanks for you help. It is driving me up the wall.

Maria DB10 (is not running in a container, but as an app in Synology)
Enable TCP/IP connection box is checked **-> checked**
Port = 3307  **-> checked**
Make sure that MariaDB port remain close from incoming connexions except from your NAS
**Reply:** dont know how i can change or verify this.

Firefly DB and User created in MariaDB using Phpmyadmin
user host = localhost
**Reply:** dont know how i can change or verify this.

Network of Firefly-iii container (Syno Docker)
Use docker host **-> checked**

Environment Variables of Firefly-iii container (Syno Docker)
DB_PORT = 3307 **-> checked**
DB_HOST = 127.0.0.1  **-> done / changed**

However it didnt work.....

I think it has something to do with the users settings, i noticed that the User name "FlireFLy dB user" with hostname "192.168.88.199" and type "database-specific" doesnt have Grant access 'No"

However this doesnt explain why it doesnt work for 127.0.01... 


![image](https://user-images.githubusercontent.com/91469589/149183099-f6073e08-5cb5-4960-b54d-3b75d5cac91d.png)

I am going to restart my NAS... maybe that helps............

**Reply by LM3469** [reply]
> Make sure that MariaDB port remain close from incoming connexions except from your NAS
Reply: dont know how i can change or verify this.

You can check this how-to if you want: https://mariushosting.com/synology-how-to-correctly-set-up-firewall-on-dsm-7/

> Firefly DB and User created in MariaDB using Phpmyadmin
user host = localhost
Reply: dont know how i can change or verify this.

You've done it: "Host name" of your MariaDB users

For me it works with:
![image](https://user-images.githubusercontent.com/97404855/149290385-8837ff3b-f110-46eb-87c1-1332dc9dffa1.png)

You may want to check the firefly dock container logs to check the errros and resolve.

**Comment by Werfjes** [comment]
Thanks LM3469, 

all done, still getting connection refused.
![image](https://user-images.githubusercontent.com/91469589/149367099-3c6f5f4f-6ac4-4b2c-af9f-ce19d851df1a.png)

**Reply by LM3469** [reply]
Well I am not sure how to help you as I shared all the info I had. Good luck :)

**Reply by MacPaille** [reply]
Still not answered my question, why not having your SQL in a docker container? It's super easy to put in place...

**Reply by Werfjes** [reply]
Hi MacPaille,

I tried the other options (they should work, only not for me). Now I will try the SQL Docker container.

However two dockers for one program sounds abit of overkill regarding resources on my NAS.

**Reply by MacPaille** [reply]
That is exactly the purpose of the containers... The have the smallest footprint with only the tools required to run your app and be completely isolated from the others.
That is what is awesome with that.

I run 4 mariadb containers for 4 different apps. And I have 4 automysqlbackup containers to do the dB backup automatically every nights.

I think you need to read more on the containers...
